DirectProgramming/DPC++FPGA/Tutorials/Tools/experimental/platform-designer/Readme.md

```diff
@@ -0,0 +1,252 @@
+   ```
+
+3. Correct the generated `_hw.tcl` file by running the `*_di_hw_tcl_adjustment_script.py` script in the generated .prj directory.
+
+   Linux/Windows:
+
+   ```
+   $> cd add.fpga_ip_export.prj_1
+   $> python add_fpga_ip_export_1_di_hw_tcl_adjustment_script.py 
+   Success! Adjusted add_fpga_ip_export_1_di_hw.tcl file!
+   The adjustment log is in: adjustments_di_hw_tcl.log
+   The original file is in: add_fpga_ip_export_1_di_hw.tcl_original
+   $> cd ..
+   $> cd add.fpga_ip_export.prj_5
+   $> python add_fpga_ip_export_5_di_hw_tcl_adjustment_script.py 
+   Success! Adjusted add_fpga_ip_export_5_di_hw.tcl file!
+   The adjustment log is in: adjustments_di_hw_tcl.log
+   The original file is in: add_fpga_ip_export_5_di_hw.tcl_original
+   ```
+
+4. **From a shell with oneAPI environment**, launch the Quartus GUI with the included Quartus project file. It's easiest to use the same shell you used to compile the `add_example` oneAPI design.
+   > **NOTE**: You may have to add a path to Quartus to your system's `PATH` environment variable. Directions for this are available in the Quartus documentation [Section 4.5: Setting Quartus Prime Environment Variables](https://www.intel.com/content/www/us/en/docs/programmable/683472/current/setting-environment-variables.html).
+
+   Linux:
+   
+   ```
+   source /opt/intel/oneapi/setvars.sh
+   ...
+   cd add-quartus
+   quartus AddCSRDemo.qpf
+   ```
+
+   Windows:
+   
+   ```
+   C:\Program Files(x86)\Intel\oneAPI\setvars.bat
+   ...
+   cd add-quartus
+   quartus.exe AddCSRDemo.qpf
+   ```
+
+> **NOTE**: You can skip steps 5-9 by running the following commands:
+> ```
+> qsys-script --script=add_kernels.tcl --quartus-project=AddCSRDemo.qpf
+> qsys-generate add_kernels.qsys --synthesis --quartus-project=AddCSRDemo.qpf --rev=add
+> quartus_sh --flow compile AddCSRDemo.qpf -c add
+> ```
+
+5. Open platform designer from the Quartus GUI:
+
+   ![](Readme.md.assets/open-platform-designer-button.png)
+
+6. Add the following IPs to your system:
+   * Basic Functions > Bridges and Adaptors > Clock > Clock Bridge Intel FPGA IP
+   * Basic Functions > Bridges and Adaptors > Reset > Reset Bridge Intel FPGA IP
+   * Basic Functions > Bridges and Adaptors > Memory Mapped > JTAG to Avalon Master Bridge Intel FPGA IP
+
+7. Add the oneAPI IPs to your system and connect them as shown:
+
+   ![](Readme.md.assets/add-ip-platform-designer.png)
+
+   ![](Readme.md.assets/complete-system_platform-designer.png)
+
+   Don't forget to export the `IRQ` and `Exception Data Bus` signals, and update the base address to prevent overlap errors.
+
+8. Save the system by clicking `File > Save`, then click `Generate HDL...` and in the window that appears, click `Generate`.
+
+   ![](Readme.md.assets/generate-hdl.png)
+
+9. When generation is complete, close platform designer, then click the 'Start Compilation' button to compile the design.
```

Comment:
Does OneAPI really need you to generate the Platform Designer system?  Most don't bother generating the RTL from Platform Designer anymore and just add the .qsys file to the Quartus project so that they just use Platform Designer to design the system and rely on Quartus to generate the Platform Designer sources and compile in one step.

Reply:
apparently it doesn't; I tried removing `qsys-generate` from the build.sh script and it still worked. I will simplify the readme.